File: petcare/lib/model-petcare/appointment.dart
import 'package:petcare/model-petcare/pet.dart';

class Appointment {
  int? appointmentId;
  int? petId;
  int? vetId;
  DateTime? appointmentDate;
  String? timeSlot;
  String? purpose;
  String? notes;
  int? serviceId;
  String? petName;

  Appointment(
      {this.appointmentId,
      this.petId,
      this.vetId,
      this.appointmentDate,
      this.timeSlot,
      this.purpose,
      this.notes,
      this.serviceId,
      this.petName});

  Appointment.fromJson(Map<String, dynamic> json) {
    appointmentId = json['appointmentId'];
    petId = json['petId'];
    vetId = json['vetId'];
    final String? dateStr = json['appointmentDate'];
    appointmentDate = dateStr != null ? DateTime.parse(dateStr) : null;
    timeSlot = json['timeSlot'];
    purpose = json['purpose'];
    notes = json['notes'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['appointmentId'] = this.appointmentId;
    data['petId'] = this.petId;
    data['vetId'] = this.vetId;
    data['appointmentDate'] = this.appointmentDate;
    data['timeSlot'] = this.timeSlot;
    data['purpose'] = this.purpose;
    data['notes'] = this.notes;
    return data;
  }
}

File: pets4life-api/BusinessObject/Models/Appointment.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace BusinessObject.Models
{
    public partial class Appointment
    {
        public int AppointmentId { get; set; }
        public int? PetId { get; set; }
        public int? VetId { get; set; }
        public DateTime? AppointmentDate { get; set; }
        public string? TimeSlot { get; set; }
        public string? Purpose { get; set; }
        public string? Notes { get; set; }

        [JsonIgnore] public virtual Pet? Pet { get; set; }
        [JsonIgnore] public virtual Veterinarian? Vet { get; set; }
    }
}
